test/GPv2Encoding.test.ts

```diff
@@ -327,6 +327,169 @@ describe("GPv2Encoding", () => {
     });
   });
 
+  describe("decodeInteraction", () => {
+    it("decodes sample encoded interaction", async () => {
+      // Example from the documentation describing `decodeInteraction` in the
+      // [`GPv2Encoding`] library.
+      const encodedInteraction =
+        "0x73c14081446bd1e4eb165250e826e80c5a523783000010000102030405060708090a0b0c0d0e0f";
+      const callData = "0x000102030405060708090a0b0c0d0e0f";
+      const target = ethers.utils.getAddress(
+        "0x73c14081446bd1e4eb165250e826e80c5a523783",
+      );
```

Comment:
Did you make all this stuff up? Does it mean that none of this stuff matters? Do all these made up values need to have a specific length?

Reply:
> Did you make all this stuff up? 

I imagine he did.

> Does it mean that none of this stuff matters?

In terms of decoding, it doesn't (as long as the format is correct, see :point_down:).

>  Do all these made up values need to have a specific length?

Yes! So, the format is described as 20 bytes for the address, followed by 3 bytes for the length `n`, followed by `n` bytes of arbitrary calldata (the contents doesn't matter, but the length must be `n`). This way, the list of encoded interactions is just one of these after the other. If the bytes are too long (extra danging bytes that isn't an interaction) or too short (no enough call data for the specified length for example) then this is an invalid interaction.

The complexity (and "weird" function signature) comes from the fact you can't say "give me the 3rd interaction" and just calculate its offset (like for trades). So the function signature becomes:
```
function decodeInteraction(
  bytes calldata encodedInteractions, // The bytes of the encoded interactions in the format :point_up: there may be one or many
  Interaction memory interaction // The location in memory where the function should write the decoded values
) returns (
  bytes calldata reducedEncodedInteractions // The leftover bytes that were not used, this will be empty for the last interaction
) 
```

Reply:
The data comes from [here](https://github.com/gnosis/gp-v2-contracts/pull/250/files#diff-ddcc995e0065433a12898489b4b70414eff5af73fd695c88572b51ee63629938R341-R346) (where I made it up in the first place). This test is there to confirm that the example I used does indeed work as expected. The example was made up mainly to explain the length: you should be able to recognize that the length is 0x10 = 16 and by how I build the calldata you should see that the length of the calldata is indeed 16 without too much effort (`0x00`, `0x01`, ..., `0x0f`)